Feature: Show UPE opt-in banner

Mayisha opened the pull request:
Fixes #1693

### Description
Adding a UPE opt-in banner in the UPE payment method pages. The new banner which is created as a React component is inserted within the old UI of the pages.

### Changes proposed in this PR
- Created a `Pill` component to show the `Early Access` tag. (styles are copied from WCPay)
- Created a `UPEOptInBanner` component for the main card
- Created separate component for `all-payment-methods` icon
- Added entry in the webpack config file exposing the `UPEOptInBanner` file path
- Added `admin_options` function in `WC_Stripe_Payment_Gateway`. This is printing the form fields and inserting an empty `div` with `id=wc-stripe-upe-ont-in-banner` if `_wcstripe_feature_upe_settings` flag is enabled. The form fields are untouched otherwise.
- Added an `admin_script` to register the generated js and css files of the new `UPEOptInBanner` component in the abstract class.
- Enqueueing the script in the constructors of each UPE payment methods class.

Desktop view:
![desktop](https://user-images.githubusercontent.com/33387139/130948204-8945365b-5d7f-462a-a403-283b8aa6db4e.png)

Mobile view:
![mobile](https://user-images.githubusercontent.com/33387139/130948226-77353e19-3123-48c3-aa4d-46273fb8524e.png)

### Testing instructions
- Enable the `_wcstripe_feature_upe_settings` flag. (One way is to simply returning `true` from [this function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/e9c04deb2771f46ad082e2cde3b45f785c160ff6/includes/class-wc-stripe-feature-flags.php#L21))
- Check all the UPE payment method pages
- Make sure the QA steps are satisfied

### QA steps

- [ ] Banner is visible in all the UPE payment method pages when feature flag is on
- [ ] Banner is absent from all the UPE payment method pages when feature flag is off
- [ ] Banner is absent from all the main Stripe page
- [ ] Desktop view supports the design
- [ ] Mobile view supports the design
- [ ] UI is responsive

Mayisha committed:
Create 'Pill' component

Mayisha committed:
Add all-payment-methods image
- Add image svg
- Create component for the image

Mayisha committed:
Create 'UPEOptInBanner' component

Mayisha committed:
Add tests for 'UPEOptInBanner' component

Mayisha committed:
Add entry in webpack config for UPE opt-in banner

Mayisha committed:
Add admin_options method to insert placeholder div for UPE opt-in banner

Mayisha committed:
Add admin script for banner
- Add a function for registering scripts in the abstract class
- Register the opt-in banner related js and css files in the script
- Enqueue the script only in the relevant UPE payment method constructors

Mayisha committed:
Fix typo

tommyshellberg commented on client/settings/pill.js:
```diff
@@ -0,0 +1,23 @@
+/**
```
We mentioned this in the meeting but it would be a good idea to have this in a /components/ folder since it will be re-usable.

tommyshellberg commented on client/settings/upe-opt-in-banner/index.js:
```diff
@@ -0,0 +1,100 @@
+			</p>
+			<Actions>
+				<span>
+					<Button isPrimary onClick={() => alert('Will allow user to enable in store')}>
```
I think this button will just link to a URL and won't need to perform a function within the app, what if we get rid of the onClick handler and just pass in a '#' for `href` as a placeholder for now? Will make it easier to update the link when we have it.

tommyshellberg commented on client/settings/upe-opt-in-banner/index.js:
```diff
@@ -0,0 +1,100 @@
+import Pill from '../pill';
+import AllPaymentMethodsIcon from '../../payment-method-icons/all-payment-methods';
+
+const BannerWrapper = styled(Card)`
```
If you use an IDE I would set up `eslint` which should be able to pick up the eslint rules from the project. There are some spacing issues that can be fixed( space after opening parenthesis and before closing parenthesis).

tommyshellberg commented on includes/abstracts/abstract-wc-stripe-payment-gateway.php:
```diff
@@ -29,6 +29,61 @@ public function display_admin_settings_webhook_description() {
+
+			wp_register_style(
+				'woocommerce_stripe_upe_opt_in',
+				plugins_url( 'build/style-upe_opt_in_banner.css', WC_STRIPE_MAIN_FILE ),
```
Do we need to import a CSS stylesheet for the banner? I don't see a stylesheet file for the OptInBanner component and it seems like the CSS is being injected directly into the components with `@emotion/styled`.

Mayisha replied:
Right! We don't need the stylesheet here. Will remove the unnecessary parts.

Mayisha committed:
Move 'Pill' inside the component folder

Mayisha committed:
Remove unnecessary code for registering style
- As there is no separate css file and the elements are wrapped with @emotion/styled for styling, registering style is not needed here

Mayisha committed:
Fix lint issues and 'Enable in your store' link

tommyshellberg commented on client/settings/upe-opt-in-banner/index.js:
```diff
@@ -0,0 +1,93 @@
+			<Actions>
+				<span>
+					<Button isPrimary href="?TODO" target="_blank">
+						{ __( 'Enable in your store', 'woocommerce-payments' ) }
```
Looks like the textdomain for the copied strings is still set to `woocommerce-payments`, you'll want to set it to `woocommerce-gateway-stripe` for any strings translated with the `__()` function

tommyshellberg replied:
I went ahead and fixed it since I found an incorrect textdomain from my own work 😆

tommyshellberg committed:
Fix textdomains.

tommyshellberg reviewed (approved):
Looks good to me! Nice job 👍

Mayisha committed:
Merge branch 'develop' into feature/1693-upe-opt-in-banner
